docker/Dockerfile

```diff
@@ -26,21 +26,31 @@ RUN echo "Install base packages" && apt-get update \
 
 COPY docker/Arial.ttf /usr/share/fonts/truetype/msttcorefonts/
 
-WORKDIR /home/vcap/app
-
-COPY policy.xml entrypoint.sh ./
+COPY policy.xml .
 
 # Overwrite the default ImageMagick policy which doesn't allow reading or writing PDFs
 RUN rm /etc/ImageMagick-6/policy.xml && cp ./policy.xml /etc/ImageMagick-6/policy.xml
+WORKDIR /home/vcap/app
+
+# Build/Install python dependencies
+FROM base AS python_build
```

Comment:
Why do we want to use the `python_build` step here? Is there any particular benefit to put it in a separate step instead of expanding the `base` step to:

1. Install all system packages
2. Create notify and celery* users
3. Install requirements as the notify user

That way we avoid copying and `chown`ing things

*Why do we need a celery user along with a notify user? wouldn't it be simpler to only have a single low-privileges user?

Reply:
> *Why do we need a celery user along with a notify user? wouldn't it be simpler to only have a single low-privileges user?

Good question - we probably don't? I just didn't think to remove it .. for some reason. Incase `celeryuser` was expected somewhere else I guess... I will remove it 👍

Reply:
Oh I hadn't realised that we were using it already 🤔 
Still, probably better to simplify anyway 😄

Reply:
Yeah, I guess that was the workaround at the time to get rid of an aggressive celery warning that is thrown if you try to run celery as root 😉 

---

Re the `python_build` step - the benefits might've been more obvious if I actually made good use of it 😂  The idea is that a number of the OS-level things we install are only required for installing/building the various python packages we need. If we build them in a separate stage, and then just copy the result, we are doing something similar to the `frontend_build` steps elsewhere and removing a load of packages from the prod build that aren't needed there. This results in a smaller attack surface (no dev tooling installed+available) and a smaller image (about 970MB -> 770MB) with change 1c013d32c93eaf69b27e50395e684cb9df6c51d0.